proto/messages.go

```diff
@@ -1257,10 +1254,13 @@ func ReadVersionedFetchResp(r io.Reader, version int16) (*FetchResp, error) {
 				return nil, dec.Err()
 			}
 
-			if msgSetSize > 0 {
+			br := bufio.NewReader(io.LimitReader(r, int64(msgSetSize)))
```

Comment:
It looks like we use LimitReader with the same limit twice.  Here and inside readRecordBatch

Reply:
Ah yes, good point.  I think we can get rid of the limit in both `readMessageSet` and `readRecordBatch`.  It's not even correct in `readRecordBatch`, since for every batch except the last, there will be more data in the reader than it consumes anyway.

`readMessageSet` still needs the `size` parameter, since it does some other things with it, but `readRecordBatch` probably doesn't need it anymore.

Reply:
Done.